examples/mini-app/src/hooks/useGameLogic.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  CASINO_GAME_TYPE,
+  COINTOSS_FACE,
+  DiceNumber,
+  chainById,
+  chainNativeCurrencyToToken,
+} from "@betswirl/sdk-core"
+import React, { useState } from "react"
+import { formatGwei } from "viem"
+import { useAccount, useBalance } from "wagmi"
+import { useChain } from "../context/chainContext"
+import { BetStatus, GameResult } from "../types"
+import { useGameCalculations } from "./useGameCalculations"
+import { HistoryEntry, useGameHistory } from "./useGameHistory"
+import { useHouseEdge } from "./useHouseEdge"
+import { usePlaceBet } from "./usePlaceBet"
+
+type GameSelection = COINTOSS_FACE | DiceNumber
+
+interface UseGameLogicProps<T extends GameSelection> {
+  gameType: CASINO_GAME_TYPE
+  defaultSelection: T
+  backgroundImage: string
+}
+
+interface UseGameLogicResult<T extends GameSelection> {
+  isWalletConnected: boolean
+  address: string | undefined
+  balance: bigint
+  tokenDecimals: number
+  areChainsSynced: boolean
+  gameHistory: HistoryEntry[]
+  refreshHistory: () => void
+  refetchBalance: () => void
+  betAmount: bigint | undefined
+  setBetAmount: (amount: bigint | undefined) => void
+  selection: T
+  setSelection: (selection: T) => void
+  betStatus: BetStatus
+  gameResult: GameResult | null
+  resetBetState: () => void
+  formattedVrfFees: number | string
+  gasPrice: string
+  targetPayoutAmount: bigint
+  multiplier: string
+  isInGameResultState: boolean
+  nativeCurrencySymbol: string
+  themeSettings: {
+    theme: "light" | "dark" | "system"
+    customTheme?: {
+      "--primary"?: string
+      "--play-btn-font"?: string
+      "--game-window-overlay"?: string
+    } & React.CSSProperties
+    backgroundImage: string
+  }
+  handlePlayButtonClick: () => void
+  handleBetAmountChange: (amount: bigint | undefined) => void
+  placeBet: (betAmount: bigint, choice: T) => void
+}
+
+export function useGameLogic<T extends GameSelection>({
+  gameType,
+  defaultSelection,
+  backgroundImage,
+}: UseGameLogicProps<T>): UseGameLogicResult<T> {
+  const { isConnected: isWalletConnected, address } = useAccount()
+  const { gameHistory, refreshHistory } = useGameHistory(gameType)
```

Comment:
The balance fetch could fail but there's no error handling. Consider wrapping in a try-catch or using the error state from useBalance:
```typescript
const { data: balance, refetch: refetchBalance, error: balanceError } = useBalance({ address })

// Handle error state appropriately
if (balanceError) {
  console.error('Failed to fetch balance:', balanceError)
}
```